webhook_to_fedora_messaging_messages/forgejo/utils.py

```diff
@@ -0,0 +1,134 @@
+# SPDX-FileCopyrightText: 2024-2025 Contributors to the Fedora Project
+#
+# SPDX-License-Identifier: LGPL-3.0-or-later
+
+
+def summarize_repository_event(event_type: str, data: dict) -> str:
+    """
+    Obtain text specification based on event type
+    """
+    if event_type == "push":
+        return _summarize_push_event(data)
+    elif event_type == "fork":
+        return _summarize_fork_event(data)
+    elif event_type == "pull_request":
+        return _summarize_pull_request_event(data)
+    elif event_type == "issues":
+        return _summarize_issues_event(data)
+    elif event_type == "issue_comment":
+        return _summarize_issue_comment_event(data)
+    else:
+        return "Event type not supported"
+
+
+def _summarize_push_event(data) -> str:
+    """
+    Obtain text specification for push event
+    """
+    repository_name = data["repository"]["full_name"]
+    repository_url = data["repository"]["html_url"]
+    pusher = data["pusher"]["full_name"]
+    pusher_url = data["pusher"]["html_url"]
+    branch = data["ref"].split("/")[-1]
+    commits = "\n".join(
+        [
+            f"Commit: {commit['message']} by {commit['author']['name']} ({commit['url']})"
+            for commit in data["commits"]
+        ]
+    )
+    return (
+        f"Event: Push\n"
+        f"Repository: {repository_name} ({repository_url})\n"
+        f"Pusher: {pusher} ({pusher_url})\n"
+        f"Branch: {branch}\n"
+        f"Commits:\n{commits}"
+    )
+
+
+def _summarize_fork_event(data) -> str:
+    """
+    Obtain text specification for fork event
+    """
+    repository_name = data["repository"]["full_name"]
+    repository_url = data["repository"]["html_url"]
+    forkee = data["forkee"]["full_name"]
+    forkee_url = data["forkee"]["html_url"]
+    owner = data["forkee"]["owner"]["full_name"]
+    owner_url = data["forkee"]["owner"]["html_url"]
+    return (
+        f"Event: Fork\n"
+        f"Repository: {repository_name} ({repository_url})\n"
+        f"Forkee: {forkee} ({forkee_url})\n"
+        f"Owner: {owner} ({owner_url})"
+    )
+
+
+def _summarize_pull_request_event(data) -> str:
+    """
+    Obtain text specification for pull request event
+    """
+    repository_name = data["repository"]["full_name"]
+    repository_url = data["repository"]["html_url"]
+    pr_title = data["pull_request"]["title"]
+    pr_url = data["pull_request"]["html_url"]
+    pr_author = data["pull_request"]["user"]["full_name"]
+    pr_source_branch = data["pull_request"]["head"]["ref"]
+    pr_dest_branch = data["pull_request"]["base"]["ref"]
+    pr_reviewers = ", ".join(
+        [reviewer["full_name"] for reviewer in data["pull_request"]["requested_reviewers"]]
```

Comment:
"full_name" can be empty, as your test example shows. It would be preferable to use something like `(reviewer["full_name"] or reviewer["login"])` here, otherwise the reviewer won't be shown.